mkosi/backend.py

```diff
@@ -948,3 +949,25 @@ def complete_step(cls, text: str, text2: Optional[str] = None) -> Iterator[List[
 
         if text2 is not None:
             cls.print_step(text2.format(*args))
+
+
+def chown_to_running_user(path: PathString) -> None:
+    uid = int(os.getenv("SUDO_UID") or os.getenv("PKEXEC_UID") or str(os.getuid()))
+    user = pwd.getpwuid(uid).pw_name
+    gid = pwd.getpwuid(uid).pw_gid
+
+    with MkosiPrinter.complete_step(
+        f"Changing ownership of output file {path} to user {user}…",
+        f"Changed ownership of {path}",
+    ):
+        os.chown(path, uid, gid)
+
+
+def mkdir_chown_current_user(path: Path, skip_chown: bool = False, mode: int = 0o777) -> None:
+    if os.path.isdir(path):
+        return
+
+    path.mkdir(mode=mode)
```

Comment:
This basically makes `exists_ok=True` the one and only behaviour. I'd prefer adding the same argument (with `True` as default) and plumbing it through.